Assets/HoloToolkit/Utilities/Scripts/Singleton.cs

```diff
@@ -8,28 +8,41 @@ namespace HoloToolkit.Unity
     /// <summary>
     /// Singleton behaviour class, used for components that should only have one instance
     /// </summary>
-    /// <typeparam name="T"></typeparam>
+    /// <typeparam name="T">The Singleton Type</typeparam>
     public class Singleton<T> : MonoBehaviour where T : Singleton<T>
     {
         private static T instance;
+
+        /// <summary>
+        /// Returns the Singleton instance of the classes type.
+        /// If no instance is found, then we search for an instance
+        /// in the scene.
+        /// If more than one instance is found, we throw an error and
+        /// no instance is returned.
+        /// </summary>
         public static T Instance
         {
             get
             {
-                if (instance == null)
+                if (instance == null && searchForInstance)
                 {
-                    Debug.LogErrorFormat(
-                        "The {0} singleton instance is being accessed while it's null. Make sure it's included in"
-                            + " the scene, and your initialization/deinitialization sequence is compatible with"
-                            + " singletons.",
-                        typeof(T).Name
-                        );
+                    searchForInstance = false;
+                    T[] objects = FindObjectsOfType<T>();
+                    if (objects.Length == 1)
+                    {
+                        instance = objects[0];
```

Comment:
if we got here, then presumably `.Instance` was called before `base.Awake()`. Maybe a warning to make sure the implementing class is calling `base.Awake()`. You'll only hit it once at runtime.

Reply:
Idk if a warning is needed.  It's could have been the uncertain script execution order that we got here before `Awake`, instead of the improperly calling `base.Awake`.  I wonder how we can definitively check for that.

Reply:
I'd imagine it would look something like what was removed from the `(instance == null)` block.
> `Debug.LogFormat(".Instance was called before .Awake(); If {0}.Awake() calls base.Awake() you can ignore this message.", typeof(T).Name);`